mammoth/translate/beam_search.py

```diff
@@ -238,6 +260,20 @@ def remove_finished_batches(self, _B_new, _B_old, non_finished, predictions, att
         self._batch_index = self._batch_index.index_select(0, non_finished)
         self.select_indices = self._batch_index.view(_B_new * self.beam_size)
         self.alive_seq = predictions.index_select(0, non_finished).view(-1, self.alive_seq.size(-1))
+        # non_finished is batch-level, and we want to apply it to all beams in the batch.
+        # encoder_output_tiled and src_mask_tiled have collapsed batches and beams into dim 0
+        self.encoder_output_tiled = rearrange(
```

Comment:
i get the logic (reshape -> select -> reshape) but this is very much not dev-friendly.
 break it down in three explicit instructions instead